_posts/2024-02-19-jco-1.0.md

```diff
@@ -0,0 +1,270 @@
+    -o target/jco                                   # Write the result to `target/jco`
+```
+
+On success it will tell you it's generated a number of JS files as part of
+its CLI output:
+
+```text
+Transpiled JS Component Files:
+
+ - target/jco/interfaces/wasi-cli-environment.d.ts      0.09 KiB
+ - target/jco/interfaces/wasi-cli-exit.d.ts             0.16 KiB
+ - target/jco/interfaces/wasi-cli-stderr.d.ts           0.17 KiB
+ - target/jco/interfaces/wasi-cli-stdin.d.ts            0.17 KiB
+ - target/jco/interfaces/wasi-cli-stdout.d.ts           0.17 KiB
+ - target/jco/interfaces/wasi-clocks-wall-clock.d.ts    0.11 KiB
+ - target/jco/interfaces/wasi-filesystem-preopens.d.ts   0.2 KiB
+ - target/jco/interfaces/wasi-filesystem-types.d.ts     2.71 KiB
+ - target/jco/interfaces/wasi-io-error.d.ts             0.08 KiB
+ - target/jco/interfaces/wasi-io-streams.d.ts           0.58 KiB
+ - target/jco/playground_jco.core.wasm                  1.83 MiB
+ - target/jco/playground_jco.core2.wasm                 16.5 KiB
+ - target/jco/playground_jco.d.ts                       0.72 KiB
+ - target/jco/playground_jco.js                           40 KiB
+```
+
+The next step is to import this into our Node.js file. In order to do this, we
+first have to update our `package.json` to declare that we're writing a JS
+module. In our package.json, add the following line:
+
+```json
+{
+  "type": "module",
+}
+```
+
+And then finally, we're ready to write some JS code. we're going to import the
+`scream` function from our Rust library, pass it a string, and then print the
+output of the string.
+
+```js
+import { scream } from "./target/jco/playground_jco.js";
+
+let msg = scream("chashu would like some tuna");
+console.log(msg);
+```
+
+Now if we run this using `node index.js`, we should get the following output:
+
+```text
+CHASHU WOULD LIKE SOME TUNA!!1!
+```
+
+And with that, we've successfully written our first Rust component for Node.js.
+You can give yourself a little round of applause - because this is *incredibly
+cool*! By leveraging WIT and Wasm, we've been able to define a language-agnostic
+interface which resulted in strongly typed code for both programming languages.
+And by leveraging code generation, all of the hard FFI and data marshalling bits
+have been handled for us - enabling us to focus on what is truly important
+(screaming we want tuna).
+
+This is just a taste of what Wasm, WASI, jco, and language toolchains can do.
+The [latest version of WASI](https://bytecodealliance.org/articles/WASI-0.2)
+(0.2.0) provides among other things access to asynchronous sockets, timers, and
+http. Using the same broad techniques we've shown in this post, it should for example be
+possible to write your own polyglot networked applications, today.
+
+## Conclusion
+
+In this post we’ve introduce jco 1.0.0: a native JavaScript runtime which can
+run WASI 0.2.0 components. We’ve discussed what has been stabilized, which
```

Comment:
See, here it's WASI 0.2.0 again. One or the other, or both to help with translations between them.